nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/util/crypto/CipherUtility.java

```diff
@@ -0,0 +1,271 @@
+import org.apache.commons.lang3.StringUtils;
+import org.apache.nifi.processor.exception.ProcessException;
+import org.apache.nifi.stream.io.ByteArrayOutputStream;
+import org.apache.nifi.stream.io.StreamUtils;
+
+import javax.crypto.Cipher;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CipherUtility {
+
+    public static final int BUFFER_SIZE = 65536;
+
+    /**
+     * Returns the cipher algorithm from the full algorithm name. Useful for getting key lengths, etc.
+     * <p/>
+     * Ex: PBEWITHMD5AND128BITAES-CBC-OPENSSL -> AES
+     *
+     * @param algorithm the full algorithm name
+     * @return the generic cipher name or the full algorithm if one cannot be extracted
+     */
+    public static String parseCipherFromAlgorithm(final String algorithm) {
+        if (StringUtils.isEmpty(algorithm)) {
+            return algorithm;
+        }
+        String formattedAlgorithm = algorithm.toUpperCase();
+
+        // This is not optimal but the algorithms do not have a standard format
+        final String AES = "AES";
+        final String TDES = "TRIPLEDES";
+        final String TDES_ALTERNATE = "DESEDE";
+        final String DES = "DES";
+        final String RC4 = "RC4";
+        final String RC2 = "RC2";
+        final String TWOFISH = "TWOFISH";
+        final List<String> SYMMETRIC_CIPHERS = Arrays.asList(AES, TDES, TDES_ALTERNATE, DES, RC4, RC2, TWOFISH);
+
+        // The algorithms contain "TRIPLEDES" but the cipher name is "DESede"
+        final String ACTUAL_TDES_CIPHER = "DESede";
+
+        for (String cipher : SYMMETRIC_CIPHERS) {
+            if (formattedAlgorithm.contains(cipher)) {
+                if (cipher.equals(TDES) || cipher.equals(TDES_ALTERNATE)) {
+                    return ACTUAL_TDES_CIPHER;
+                } else {
+                    return cipher;
+                }
+            }
+        }
+
+        return algorithm;
+    }
+
+    /**
+     * Returns the cipher key length from the full algorithm name. Useful for getting key lengths, etc.
+     * <p/>
+     * Ex: PBEWITHMD5AND128BITAES-CBC-OPENSSL -> 128
+     *
+     * @param algorithm the full algorithm name
+     * @return the key length or -1 if one cannot be extracted
+     */
+    public static int parseKeyLengthFromAlgorithm(final String algorithm) {
+        int keyLength = parseActualKeyLengthFromAlgorithm(algorithm);
+        if (keyLength != -1) {
+            return keyLength;
+        } else {
+            // Key length not explicitly named in algorithm
+            String cipher = parseCipherFromAlgorithm(algorithm);
+            return getDefaultKeyLengthForCipher(cipher);
+        }
+    }
+
+    private static int parseActualKeyLengthFromAlgorithm(final String algorithm) {
+        Pattern pattern = Pattern.compile("([\\d]+)BIT");
```

Comment:
Would recommend defining the Pattern as a 'private static final' member variable, as it is thread-safe and doesn't change